frame/composable-support/src/abstractions/block_fold.rs

```diff
@@ -0,0 +1,119 @@
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use frame_support::{
+	pallet_prelude::{OptionQuery, StorageMap, StorageValue},
+	storage::types::QueryKindTrait,
+	traits::{Get, StorageInstance},
+	ReversibleStorageHasher, StorageHasher,
+};
+use scale_info::TypeInfo;
+
+pub trait FoldStorage<S, K, V> {
```

Comment:
What is the use of this and where is it used? Just curious

IMO Would be nice to have some docs for understanding and tests to verify the code

Reply:
Folding over a storage, block per block, in a chunk fashion.

Reply:
[Used here for instance](https://github.com/ComposableFi/composable/blob/d114b7fac4dea3da0784b09467cc835b2fccfda4/frame/staking-rewards/src/lib.rs#L426)

Reply:
This avoid having to manually track the folding state/iteration key and make you focus on the actual lambda that fold